spec/inputs/gelf_spec.rb

```diff
@@ -146,4 +146,41 @@
       expect(event.timestamp.usec).to be_within(1000).of(123456)
     end
   end
+
+  context "when an invalid JSON is fed to the listener" do
+    subject { LogStash::Inputs::Gelf.new_event(message, "host") }
+    let(:message) { "Invalid JSON message" }
+
+    context "legacy JSON parser output" do
+      before(:each) do
+        class << LogStash::Inputs::Gelf; alias_method(:parse, :legacy_parse); end
```

Comment:
@colinsurprenant Do you know a better way to do this?

I tried `allow(LogStash::Event).to respond_to(:from_json) { true/false }` but it didn't work as expected.